### PR TITLE
tabnine: fix binary permissions

### PR DESCRIPTION
**Pull request by onsails**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/tabnine/default.nix [hunk 1]
```diff
@@ -24,6 +24,7 @@ in stdenv.mkDerivation rec {
 
   installPhase = ''
     mkdir -p $out/bin
+    chmod +x $src
     ln -s $src $out/bin/TabNine
```
Comment (SuperSandro2000) [inline pkgs/development/tools/tabnine/default.nix:28]:
```suggestion
    install -Dm755 $src $out/bin/TabNine
```
This should be easier. Please test.

**Comment by onsails** [comment]
Result of `nixpkgs-review pr 105149` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>tabnine</li>
    <li>vimPlugins.completion-tabnine</li>
  </ul>
</details>

File: pkgs/development/tools/tabnine/default.nix (hunk 1 above)
Comment (onsails) [reply pkgs/development/tools/tabnine/default.nix]: it works, thanks

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105149` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tabnine</li>
  </ul>
</details>

**Pull request merged**